### PR TITLE
Debug display/motor ui

### LINKED ISSUE #65: Motor Ui
1) Make 4x5 Table showing important information of Left Right as well as an Avg values of the two
2) Make a separate 4xXX to show all the rest of the info, as well as a scrollbar

### PR DESCRIPTION
**Pull request by safasyed**
Finished motor Ui and motor view files 

Closes #65

### CODE REVIEW

**Comment by bill-luu** [comment]
Fix up the merge conflicts!

**Comment by bill-luu** [comment]
Also, going forward put your name in front of your branch; so something like `safa/DebugDisplay/MotorUI`

File: src/ViewLayer/ViewContainer.h [hunk 1]
```diff
@@ -37,7 +38,11 @@ class ViewContainer
     MotorUi* motorUi_;
     MpptUi* mpptUi_;
     TabUi* tabUi_;
+<<<<<<< HEAD:ViewLayer/ViewContainer.h
```
Comment (bill-luu) [inline src/ViewLayer/ViewContainer.h]: These still need to be removed

File: src/ViewLayer/ViewContainer.h [hunk 2]
```diff
@@ -37,7 +38,11 @@ class ViewContainer
...
+    QScopedPointer<MotorView> MotorView_;
+=======
```
Comment (bill-luu) [inline src/ViewLayer/ViewContainer.h]: Remove

File: src/ViewLayer/ViewContainer.h [hunk 3]
```diff
@@ -37,7 +38,11 @@ class ViewContainer
...
     QScopedPointer<ControlView> ControlView_;
+>>>>>>> 2247104d376b5700a6d5f2bd8be91acade236c91:src/ViewLayer/ViewContainer.h
```
Comment (bill-luu) [inline src/ViewLayer/ViewContainer.h]: Remove

File: src/ViewLayer/ViewLayer.pro [hunk 4]
```diff
@@ -27,11 +27,16 @@ HEADERS += \
     DebugDisplay/Tab/TabUi/I_TabUi.h \
     DebugDisplay/MPPTPage/MPPTUi/MpptUi.h \
     DebugDisplay/MPPTPage/MPPTUi/I_MpptUi.h \
+<<<<<<< HEAD:ViewLayer/ViewLayer.pro
```
Comment (bill-luu) [inline src/ViewLayer/ViewLayer.pro]: Remove

File: src/ViewLayer/ViewLayer.pro [hunk 5]
```diff
@@ -27,11 +27,16 @@ HEADERS += \
...
+    DebugDisplay/MPPTPage/MPPTView/MpptView.h \
+    DebugDisplay/MotorPage/MotorUi/MotorView.h
+=======
```
Comment (bill-luu) [inline src/ViewLayer/ViewLayer.pro]: Remove

File: src/ViewLayer/ViewLayer.pro [hunk 6]
```diff
@@ -27,11 +27,16 @@ HEADERS += \
...
     DebugDisplay/ControlPage/ControlView/ControlView.h \
     DebugDisplay/MPPTPage/MPPTView/MpptView.h \
     RaceModeDisplay/RaceModeDashboardView.h \
     RaceModeDisplay/RaceModeDisplayUI/I_RaceModeDashboardUI.h \
     RaceModeDisplay/RaceModeDisplayUI/RaceModeDashboardUI.h
+>>>>>>> 2247104d376b5700a6d5f2bd8be91acade236c91:src/ViewLayer/ViewLayer.pro
```
Comment (bill-luu) [inline src/ViewLayer/ViewLayer.pro]: Remove

File: src/ViewLayer/ViewLayer.pro [hunk 7]
```diff
@@ -45,10 +50,15 @@ SOURCES += \
     DebugDisplay/MotorPage/MotorUi/MotorUi.cpp \
     DebugDisplay/Tab/TabUi/TabUi.cpp \
     DebugDisplay/MPPTPage/MPPTUi/MpptUi.cpp \
+<<<<<<< HEAD:ViewLayer/ViewLayer.pro
```
Comment (bill-luu) [inline src/ViewLayer/ViewLayer.pro]: Remove

File: src/ViewLayer/ViewLayer.pro [hunk 8]
```diff
@@ -45,10 +50,15 @@ SOURCES += \
...
+    DebugDisplay/MPPTPage/MPPTView/MpptView.cpp \
+    DebugDisplay/MotorPage/MotorUi/MotorView.cpp
+=======
```
Comment (bill-luu) [inline src/ViewLayer/ViewLayer.pro]: Remove

File: src/ViewLayer/ViewLayer.pro [hunk 9]
```diff
@@ -45,10 +50,15 @@ SOURCES += \
...
     DebugDisplay/ControlPage/ControlView/ControlView.cpp \
     DebugDisplay/MPPTPage/MPPTView/MpptView.cpp \
     RaceModeDisplay/RaceModeDashboardView.cpp \
     RaceModeDisplay/RaceModeDisplayUI/RaceModeDashboardUI.cpp
+>>>>>>> 2247104d376b5700a6d5f2bd8be91acade236c91:src/ViewLayer/ViewLayer.pro
```
Comment (bill-luu) [inline src/ViewLayer/ViewLayer.pro]: Remove

File: ViewLayer/DebugDisplay/MotorPage/MotorUi/MotorView.cpp [hunk 10]
```diff
@@ -0,0 +1,374 @@
...
+{
+}
+
+void MotorView::connectMotor(KeyMotorPresenter& keyMotorPresenter, MotorDetailsPresenter& motorDetailsPresenter)
+{
+    connect(&keyMotorPresenter, SIGNAL(motorZeroReceived(KeyMotor)),
+            this, SIGNAL(motorZeroReceived(KeyMotor)));
+    connect(&keyMotorPresenter, SIGNAL(motorOneReceived(KeyMotor)),
+            this, SIGNAL(motorOneReceived(KeyMotor)));
+    connect(&keyMotorPresenter, SIGNAL(motorSetCurrentReceived(double setCurrent)),
```
Comment (bill-luu) [inline ViewLayer/DebugDisplay/MotorPage/MotorUi/MotorView.cpp]: For signals, let's keep things consistent and do `SIGNAL(motorSetCurrentReceived(double)`
Actually I think we need to do it like that, or else the signals won't work properly.
This comment goes for the rest of them.

File: ViewLayer/DebugDisplay/MotorPage/MotorUi/motorview.h.autosave [hunk 11]
```diff
@@ -0,0 +1,25 @@
+#pragma once
```
Comment (bill-luu) [inline ViewLayer/DebugDisplay/MotorPage/MotorUi/motorview.h.autosave]: Remove this autosave file.

File: src/ViewLayer/DebugDisplay/MotorPage/MotorUi/MotorView.cpp [hunk 12]
```diff
@@ -0,0 +1,374 @@
...
+    connect(&keyMotorPresenter, SIGNAL(motorBusCurrentReceived(double busCurrent)),
+            this, SIGNAL(motorBusCurrentReceived(double busCurrent)));
+
+    connect(&motorDetailsPresenter, SIGNAL(motorZeroDetailsReceived(MotorDetails)),
+            this, SIGNAL(motorZeroDetailsReceived(MotorDetails)));
+    connect(&motorDetailsPresenter, SIGNAL(motorOneDetailsReceived(MotorDetails)),
+            this, SIGNAL(motorOneDetailsReceived(MotorDetails)));
+}
+
+
```
Comment (bill-luu) [inline src/ViewLayer/DebugDisplay/MotorPage/MotorUi/MotorView.cpp]: Extra new line

**Comment by bill-luu** [comment]
Fix merge conflicts/

**Comment by bill-luu** [comment]
Still need to format

File: src/ViewLayer/DebugDisplay/MotorPage/MotorUi/MotorView.cpp (hunk 12 above)
Comment (bill-luu) [reply src/ViewLayer/DebugDisplay/MotorPage/MotorUi/MotorView.cpp]: Still needs to be addressed

File: src/ViewLayer/DebugDisplay/MotorPage/MotorView/MotorView.cpp [hunk 13]
```diff
@@ -0,0 +1,374 @@
...
+    ui_.slipSpeedRightLabel().setNum(slipSpeedRight);
+}
+
+void MotorView::slipSpeedAvg(double slipSpeedLeft, double slipSpeedRight)
+{
+    double slipSpeedAvg = (slipSpeedLeft + slipSpeedRight) / 2;
+    ui_.slipSpeedAvgLabel().setNum(slipSpeedAvg);
+}
+
+
```
Comment (bill-luu) [inline src/ViewLayer/DebugDisplay/MotorPage/MotorView/MotorView.cpp]: Extra whitespace

File: ViewLayer/DebugDisplay/MotorPage/MotorUi/MotorView.cpp [hunk 14]
```diff
@@ -0,0 +1,374 @@
...
+    ui_.voltageRail1VSupplyRightLabel().setNum(voltageRail1VSupplyRight);
+}
+
+void MotorView::voltageRail1VSupplyAvg(double voltageRail1VSupplyLeft, double voltageRail1VSupplyRight)
+{
+    double voltageRail1VSupplyAvg = (voltageRail1VSupplyLeft + voltageRail1VSupplyRight) / 2;
+    ui_.voltageRail1VSupplyAvgLabel().setNum(voltageRail1VSupplyAvg);
+}
+
+
```
Comment (bill-luu) [inline ViewLayer/DebugDisplay/MotorPage/MotorUi/MotorView.cpp]: extra whitespace

File: ViewLayer/DebugDisplay/MotorPage/MotorUi/MotorView.cpp (hunk 10 above)
Comment (bill-luu) [reply ViewLayer/DebugDisplay/MotorPage/MotorUi/MotorView.cpp]: This comment needs to be addressed

**Comment by bill-luu** [comment]
A recent PR changed all instances of `BackEmfReal` to be `BackEmf`. Fix the merge conflicts and get that working.

**Comment by bill-luu** [comment]
Nice catches! Found some errors here and in the telemetry test tool from it

File: src/ViewLayer/DebugDisplay/MotorPage/MotorUi/MotorUi.cpp [hunk 15]
```diff
@@ -1,17 +1,332 @@
...
+}
+
+QLabel& MotorUi::setCurrentAvgLabel()
+{
+    return *ui_->setCurrentAvgLabel;
+}
+
+QLabel& MotorUi::setCurrentRightLabel()
+{
+    return *ui_->setCurrentLeftLabel;
```
Comment (bill-luu) [inline src/ViewLayer/DebugDisplay/MotorPage/MotorUi/MotorUi.cpp]: Wrong label!

**Comment by bill-luu** [comment]
The `slipCurrentAvg` label has some properties that are not like the others,like the stylesheet, font, and size, so that's probably why it's whack.

File: src/ViewLayer/DebugDisplay/MotorPage/MotorView/MotorView.cpp [hunk 16]
```diff
@@ -0,0 +1,214 @@
...
+    dcBusAmpHoursRight_ = motorOneDetails.dcBusAmpHours();
+    odometerRight_ = motorOneDetails.odometer();
+    slipSpeedRight_ = motorOneDetails.slipSpeed();
+
+
+    motorDetailsAverage();
+
+}
+
+void MotorView::motorDetailsAverage()
```
Comment (bill-luu) [inline src/ViewLayer/DebugDisplay/MotorPage/MotorView/MotorView.cpp]: Let's rename this to `setMotorDetailsAverageLabels`

File: src/ViewLayer/DebugDisplay/MotorPage/MotorView/MotorView.h [hunk 17]
```diff
@@ -0,0 +1,80 @@
...
+{
+    Q_OBJECT
+public:
+    MotorView(KeyMotorPresenter& keyMotorPresenter, MotorDetailsPresenter& motorDetailsPresenter,
+              I_MotorUi& ui);
+    ~MotorView();
+
+private:
+    void connectMotor(KeyMotorPresenter&, MotorDetailsPresenter&);
+    //void connectMotorDetails(MotorDetailsPresenter&);
```
Comment (bill-luu) [inline src/ViewLayer/DebugDisplay/MotorPage/MotorView/MotorView.h]: Remove this comment

File: src/ViewLayer/DebugDisplay/MotorPage/MotorView/MotorView.h [hunk 18]
```diff
@@ -0,0 +1,80 @@
...
+    void motorZeroReceived(KeyMotor);
+    void motorOneReceived(KeyMotor);
+    void motorSetCurrentReceived(double);
+    void motorActualSpeedReceived(double);
+    void motorBusVoltageReceived(double);
+    void motorBusCurrentReceived(double);
+
+    void motorZeroDetailsReceived(MotorDetails);
+    void motorOneDetailsReceived(MotorDetails);
+    //void motorZeroAverage(MotorDetails);
```
Comment (bill-luu) [inline src/ViewLayer/DebugDisplay/MotorPage/MotorView/MotorView.h]: Remove this comment

File: src/ViewLayer/ViewContainer.cpp [hunk 19]
```diff
@@ -58,10 +59,18 @@ ViewContainer::ViewContainer(PresenterContainer& presenterContainer, Mode mode)
                               *homepageUi_, *faultUi_,
                               *motorUi_, *mpptUi_, *tabUi_));
 
+
+//        MotorFaultView_.reset(new MotorFaultView(presenterContainer.motorFaultsPresenter(),
```
Comment (bill-luu) [inline src/ViewLayer/ViewContainer.cpp]: Remove the comment

File: src/ViewLayer/ViewContainer.h [hunk 20]
```diff
@@ -39,8 +40,10 @@ class ViewContainer
     MpptUi* mpptUi_;
     TabUi* tabUi_;
 
-    QScopedPointer<FaultView> FaultView_;
 
+    QScopedPointer<MotorView> MotorView_;
+    //QScopedPointer<MotorFaultView> MotorFaultView_;
```
Comment (bill-luu) [inline src/ViewLayer/ViewContainer.h]: Remove the comment

Comment (bill-luu) [reply src/ViewLayer/ViewContainer.h]: Still this comment

**Comment by louisj381** [comment]
Tests checked out, everything looks good.

**Pull request merged**